src/Microsoft.AspNetCore.Hosting.Abstractions/IApplicationLifetimeEvents.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Hosting
+{
+    /// <summary>
+    /// Allows consumers to perform cleanup during a graceful shutdown.
+    /// </summary>
+    public interface IApplicationLifetimeEvents
```

Comment:
Should this be an abstract base class? If we ever add events we might need to evolve this. On the other hand, it should probably match the `IApplicationLifetime`.